### PR TITLE
feat(tax_rates): Rename tax_rate into tax

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  [https://getlago.canny.io/feature-requests/p/create-several-tax-rates](https://getlago.canny.io/feature-requests/p/create-several-tax-rates)

## Context

Currently, tax can be set individually either on the customer or the organization level. However, it’s not possible to define a global tax tag that can be reusable everywhere.

## Description

The goal of this PR is to rename `tax_rates` into `taxes`

### CODE REVIEW

File: spec/requests/api/v1/taxes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::TaxesController, type: :request do
```
Comment (jdenquin) [inline spec/requests/api/v1/taxes_spec.rb:5]: RSpec/FilePath: Spec path should end with `api/v1/taxes_controller*_spec.rb`. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/FilePath)

File: db/migrate/20230510113501_create_customers_tax_rates.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 class CreateCustomersTaxRates < ActiveRecord::Migration[7.0]
   def change
-    add_column :tax_rates, :applied_by_default, :boolean, null: false, default: false
+    add_column :tax_rates, :applied_to_organization, :boolean, null: false, default: false
```
Comment (vincent-pochet) [inline db/migrate/20230510113501_create_customers_tax_rates.rb:5]: I'm not sure you want to touch this migration :)

File: db/migrate/20230522093423_rename_tax_rates_to_taxes.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class RenameTaxRatesToTaxes < ActiveRecord::Migration[7.0]
+  def change
+    rename_table :tax_rates, :taxes
+    rename_table :applied_tax_rates, :applied_taxes
+
+    rename_column :taxes, :applied_to_organization, :applied_to_organization
```
Comment (vincent-pochet) [inline db/migrate/20230522093423_rename_tax_rates_to_taxes.rb:8]:
```suggestion
    rename_column :taxes, :applied_by_default, :applied_to_organization
```

**Review by vincent-pochet — APPROVED** [review]
I sure it's alright! :p

File: db/migrate/20230510113501_create_customers_tax_rates.rb (hunk 2 above)
Comment (rsempe) [reply db/migrate/20230510113501_create_customers_tax_rates.rb:5]: thanks :)

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**